src/EditorFeatures/Core/Shared/Tagging/Utilities/TagSpanIntervalTree.cs

```diff
@@ -21,66 +21,53 @@ namespace Microsoft.CodeAnalysis.Editor.Shared.Tagging;
 {
-    private readonly ITextBuffer _textBuffer = textBuffer;
-    private readonly SpanTrackingMode _spanTrackingMode = trackingMode;
-    private readonly IntervalTree<TagSpan<TTag>> _tree = IntervalTree.Create(
-        new IntervalIntrospector(textBuffer.CurrentSnapshot, trackingMode),
-        values1, values2);
-
-    private static SnapshotSpan GetTranslatedSpan(
-        TagSpan<TTag> originalTagSpan, ITextSnapshot textSnapshot, SpanTrackingMode trackingMode)
+    // Tracking mode passed in here doesn't matter (since the tree is empty).
+    public static readonly TagSpanIntervalTree<TTag> Empty = new(SpanTrackingMode.EdgeInclusive);
+
+    private readonly SpanTrackingMode _spanTrackingMode = spanTrackingMode;
+    private readonly IntervalTree<TagSpan<TTag>> _tree = IntervalTree<TagSpan<TTag>>.Empty;
+
+    public TagSpanIntervalTree(
+        ITextSnapshot textSnapshot,
+        SpanTrackingMode trackingMode,
+        IEnumerable<TagSpan<TTag>>? values1 = null,
+        IEnumerable<TagSpan<TTag>>? values2 = null)
+        : this(trackingMode)
     {
-        var localSpan = originalTagSpan.Span;
-
-        return localSpan.Snapshot == textSnapshot
-            ? localSpan
-            : localSpan.TranslateTo(textSnapshot, trackingMode);
+        _tree = IntervalTree.Create(
+            new IntervalIntrospector(textSnapshot, trackingMode),
+            values1, values2);
     }
 
-    private TagSpan<TTag> GetTranslatedITagSpan(TagSpan<TTag> originalTagSpan, ITextSnapshot textSnapshot)
-        // Avoid reallocating in the case where we're on the same snapshot.
-        => originalTagSpan.Span.Snapshot == textSnapshot
-            ? originalTagSpan
-            : GetTranslatedTagSpan(originalTagSpan, textSnapshot, _spanTrackingMode);
+    private static SnapshotSpan GetTranslatedSpan(TagSpan<TTag> originalTagSpan, ITextSnapshot textSnapshot, SpanTrackingMode trackingMode)
+        // SnapshotSpan no-ops if you pass it the same snapshot that it is holding onto.
+        => originalTagSpan.Span.TranslateTo(textSnapshot, trackingMode);
+
+    private TagSpan<TTag> GetTranslatedTagSpan(TagSpan<TTag> originalTagSpan, ITextSnapshot textSnapshot)
+        => GetTranslatedTagSpan(originalTagSpan, textSnapshot, _spanTrackingMode);
 
     private static TagSpan<TTag> GetTranslatedTagSpan(TagSpan<TTag> originalTagSpan, ITextSnapshot textSnapshot, SpanTrackingMode trackingMode)
         // Avoid reallocating in the case where we're on the same snapshot.
-        => originalTagSpan is TagSpan<TTag> tagSpan && tagSpan.Span.Snapshot == textSnapshot
-            ? tagSpan
+        => originalTagSpan.Span.Snapshot == textSnapshot
+            ? originalTagSpan
             : new(GetTranslatedSpan(originalTagSpan, textSnapshot, trackingMode), originalTagSpan.Tag);
 
-    public ITextBuffer Buffer => _textBuffer;
-
-    public SpanTrackingMode SpanTrackingMode => _spanTrackingMode;
```

Comment:
removed these two properties.  they are not needed or appropriate to expose off of this type.